### DISCUSSION TITLE
How to deploy using feature flags?

### DISCUSSION THREAD

**Discussion by kaihendry** [serverless/serverless#10355] [Q&A] [answered]
https://wb.serverless.com/guide-ci-cd talks about using feature flags when deploying new frequent releases in serverless.

But how is it done?

Is it best achieved with AppConfig as this blog suggests https://levelup.gitconnected.com/serverless-feature-flags-6e49d534e79f ?

Or is there some native way I'm missing?

**Comment by mnapoli** [comment, marked as answer]
Hi! One simple way to have a feature flag (on/off) is to deploy it behind an environment variable/parameter. When you want to enable the feature, redeploy with the variable to `true` instead of `false`.

If you want to go further than this and expose the feature to a limited set of users (e.g. 10% of the traffic), you can use more advanced services like LaunchDarkly, or:

- AppConfig
- the newly announced CloudWatch Evidently

**Reply by kaihendry** [reply]
Yes, I think the env variable is too inflexible because of the redeployment which is a slow `sls deploy`, right? There is no quick way to turn it on and off right in the console?

**Reply by kaihendry** [reply]
I think I'll lean towards appconfig https://twitter.com/edjgeek/status/1470793386563190800
